pubnub/models/subscription.py

```diff
@@ -83,64 +84,59 @@ def unsubscribe(self):
 
 
 class PubNubSubscription(PNEventEmitter, PNSubscribeCapable):
-    def __init__(self, pubnub_instance,
-                 subscription_names: List[str],
-                 subscription_type: PNSubscriptionType) -> None:
-
+    def __init__(self, pubnub_instance, name: str, type: PNSubscriptionType, with_presence: bool = False) -> None:
         self.subscription_registry = pubnub_instance._subscription_registry
         self.subscription_manager = pubnub_instance._subscription_manager
-        self.subscription = None
-        self.subscription_names = subscription_names
-        self.subscription_type = subscription_type
+        self.name = name
+        self._type = type
+        self.with_presence = with_presence
 
     def add_listener(self, listener):
         self.subscription_registry.add_listener(listener)
 
+    def get_names_with_presence(self):
+        return [self.name, f'{self.name}-pnpres'] if self.with_presence else [self.name]
+
 
 class PubNubSubscriptionSet(PNEventEmitter, PNSubscribeCapable):
-    def __init__(self, pubnub_instance,
-                 channels: Optional[List[str]] = None,
-                 channel_groups: Optional[List[str]] = None) -> None:
-        self.subscription_type = PNSubscriptionType.SUBSCRIPTION_SET
+    _type = PNSubscriptionType.SUBSCRIPTION_SET
```

Comment:
Where do you use such an enumeration case like `PNSubscriptionType.SUBSCRIPTION_SET`?

Reply:
Good catch. Leftover from before updates